lib/resty/openssl/x509/csr.lua

```diff
@@ -162,6 +164,46 @@ function _M:to_PEM()
   return tostring(self, "PEM")
 end
 
+--- Get all csr extensions
+-- @tparam table self Instance of csr
+-- @treturn List of parsed extension objects
+function _M.get_extensions(self)
+    local extensions = C.X509_REQ_get_extensions(self.ctx)
+    local n = C.OPENSSL_sk_num(extensions)
+    local ret = {}
+    for i = 0, n - 1 do
+        local ext = C.OPENSSL_sk_value(extensions, i)
+        local dup, err = extension_lib.dup(ffi_cast(ext_typ_ptr, ext))
+        if not err then
+            local obj = dup:get_object()
+            obj.blob = dup:tostring()
+            push(ret, obj)
+        end
+    end
+    return ret
+end
+
+--- Get a csr extension
+-- @tparam table self Instance of csr
+-- @tparam string|number Nid number or name of the extension
+-- @treturn Parsed extension object or nil if not found
+function _M.get_extension(self, nid)
+    local i, err = txtnid2nid(nid)
+    if err then
+        return nil, err
+    end
+    local ctx = C.X509V3_EXT_i2d(i, 0, self.ctx)
+    if ctx == nil then
+        return nil, format_error("csr.get_extension: X509V3_EXT_i2d")
+    end
+    ffi_gc(ctx, C.X509_EXTENSION_free)
+    local dup = extension_lib.dup(ctx)
+    local obj = dup:get_object()
```

Comment:
is there a reason we want to return the parsed object instead of extension itself here?

Reply:
Yes, I my case i need blob and nid

Reply:
Yes I need blob and nid for my case

Reply:
how about just return it as x509/extension.lua instance, and caller can further call :get_object()? i'm thinking this because we usually return the structure returned by C API to be close as original API as possible

Reply:
get_object does not return blob
user has to make two calls

Reply:
let's still return x509.extension directly here, the library would be better to be usage case neutral and follow the same return type as C APi

Reply:
I had some issues with above code. Please see if you can fix it